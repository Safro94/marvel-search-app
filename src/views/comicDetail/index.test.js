/* eslint-disable import/first */
import React from 'react';
import { render, wait } from '@testing-library/react';

import ComicDetail from './';

jest.mock('../../services/service', () => {
  return {
    GetComicById: jest.fn().mockImplementation(() => {
      return {
        images: [{ path: 'path', extension: 'jpg' }],
        title: 'Spider-man',
        creators: {
          items: [
            { role: 'writter', name: 'Stan Lee' },
            { role: 'producer', name: 'Some name' },
          ],
        },
        description:
          'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Porro quibusdam nisi voluptates recusandae nesciunt quis possimus exercitationem pariatur? Officia vero voluptatem sint tempora magni ullam repudiandae fugit architecto illum molestiae!',
      };
    }),
  };
});
import Service from '../../services/service';

jest.mock('../../components/shared/image', () => 'image');

jest.mock('./style', () => {
  return {
    Wrapper: 'wrapper',
    ImageContainer: 'imageContainer',
    H1: 'h1',
    InformationContainer: 'informationContainer',
    H4: 'h4',
    Paragraph: 'paragraph',
    CreatorsInformation: 'creatorsInformation',
  };
});

describe('<ComicDetail /> ', () => {
  beforeEach(() => jest.clearAllMocks());
  const id = 30;

  it('It should render a text if there is no Id given', async () => {
    // Act
    const { container } = render(<ComicDetail.WrappedComponent match={{ params: {} }} />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('It should render a comic if we pass an Id', async () => {
    // Act
    const { container } = render(<ComicDetail.WrappedComponent match={{ params: { id: id } }} />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('If there is an Id, it should call GetComicById', async () => {
    // Act
    render(<ComicDetail.WrappedComponent match={{ params: { id: id } }} />);
    await wait();

    // Assert
    expect(Service.GetComicById).toHaveBeenCalledTimes(1);
  });
});
