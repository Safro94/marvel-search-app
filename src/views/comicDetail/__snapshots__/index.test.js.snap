// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicDetail />  It should render a comic if we pass an Id 1`] = `
<div>
  <wrapper>
    <imagecontainer>
      <image
        src="path/portrait_uncanny.jpg"
      />
    </imagecontainer>
    <informationcontainer>
      <h1>
        Spider-man
      </h1>
      <creatorsinformation>
        <h4>
          Writter: Stan Lee
        </h4>
        <h4>
          Producer: Some name
        </h4>
      </creatorsinformation>
      <paragraph>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Porro quibusdam nisi voluptates recusandae nesciunt quis possimus exercitationem pariatur? Officia vero voluptatem sint tempora magni ullam repudiandae fugit architecto illum molestiae!
      </paragraph>
    </informationcontainer>
  </wrapper>
</div>
`;

exports[`<ComicDetail />  It should render a text if there is no Id given 1`] = `
<div>
  <h1>
    No se han encontrado resultados
  </h1>
</div>
`;
